srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRuleController.java

```diff
@@ -0,0 +1,163 @@
+package gov.va.med.srcalc.web.controller.admin;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+import gov.va.med.srcalc.domain.model.Rule;
+import gov.va.med.srcalc.service.AdminService;
+import gov.va.med.srcalc.service.DuplicateRuleNameException;
+import gov.va.med.srcalc.service.InvalidIdentifierException;
+import gov.va.med.srcalc.util.ValidationCodes;
+import gov.va.med.srcalc.web.view.Views;
+import gov.va.med.srcalc.web.view.admin.EditExistingRule;
+import gov.va.med.srcalc.web.view.admin.EditRuleFactory;
+import gov.va.med.srcalc.web.view.admin.ValueMatcherBuilder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Controller;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.ModelAndView;
+
+/**
+ * Web MVC controller for editing existing rules.
+ */
+@Controller
+@RequestMapping("/admin/rules/{ruleId}")
```

Comment:
I recently learned this myself, but if you make this path a constant, MockMVC can still use it.